client/evilwallet/evilwallet.go

```diff
@@ -686,43 +687,96 @@ func (e *EvilWallet) updateOutputIDs(txID ledgerstate.TransactionID, outputs led
 }
 
 func (e *EvilWallet) PrepareTransaction(scenario *EvilScenario) (tx *ledgerstate.Transaction, err error) {
-	wallet, err := e.wallets.FreshWallet()
+	inWallet, err := e.wallets.GetNextWallet(Fresh)
```

Comment:
maybe we should change FreshWalet to lowercase, as I did the same mistake that you corrected  on my new branch with ReuseWallet 😄

Reply:
I'm confused, which do you prefer, `freshWallet()` or `GetNextWallet` with type as a parameter?

Reply:
GetNextWallet, I just forgot about it